x-pack/plugins/saved_objects_tagging/common/validation.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { Tag } from './types';
+
+export const tagNameMinLength = 2;
+export const tagNameMaxLength = 50;
+export const tagDescriptionMaxLength = 100;
+
+const hexColorRegexp = /^#[0-9A-F]{6}$/i;
+const nameValidCharsRegexp = /^[0-9A-Z:\-_\s]+$/i;
+
+export interface TagValidation {
+  valid: boolean;
+  warnings: string[];
+  errors: Partial<Record<keyof Tag, string | undefined>>;
+}
+
+const isHexColor = (color: string): boolean => {
+  return hexColorRegexp.test(color);
+};
+
+export const validateTagColor = (color: string): string | undefined => {
+  if (!isHexColor(color)) {
+    return i18n.translate('xpack.savedObjectsTagging.validation.color.errorInvalid', {
```

Comment:
I'm not very familiar with our i18n, but I think this will also translate these validation errors on the server, so if Kibana's locale is set to japanese the messages will also be japanese.

I wonder if the API _shouldn't_ be localized so that an integration doesn't break if the locale is changed? We could either only translate it on the browser, or let the API return both the english and the locale message.

Reply:
>  so if Kibana's locale is set to japanese the messages will also be japanese.

That's right.

> I wonder if the API shouldn't be localized so that an integration doesn't break if the locale is changed?

What do you call an integration? FTR tests? They are run with `en` locale. Our U/I test are not initializing i18nloader, so it would fallback to `en` (default messages) the same way.

>  or let the API return both the english and the locale message.

Let's not overcomplexity the API here imho. This is a user-facing message, it should be localized. That's not the first time, we got other APIs that are returning translated error messages, when they are user-facing.

Reply:
With integrations I was thinking of a public HTTP API that external developers could use. I missed that we wrap these messages in a `TagValidationError` with the message 'Error validating tag attributes' which doesn't gets translated, so :+1: